specification/communication/data-plane/Microsoft.CommunicationServicesTurn/preview/2021-02-22-preview1/CommunicationTurn.json

```diff
@@ -0,0 +1,180 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CommunicationIdentityClient",
+    "description": "Azure Communication Relay Service",
+    "version": "2021-02-22-preview1"
+  },
+  "paths": {
+    "/turn/{id}/:issueCredentials": {
+      "post": {
+        "tags": [
+          "Turn"
+        ],
+        "summary": "Issue TURN credentials for an existing identity.",
+        "operationId": "CommunicationIdentity_IssueTurnCredentials",
```

Comment:
The `CommunicationIdentity.json` swagger already defines a class called `CommunicationIdentity`.  So you'll have the same class name in two different packages which seems odd.  Can you please clarify the intent here.

Reply:
Not sure I follow - I'm not defining the `CommunicationIdentity` class in this swagger. I have the operationId under `CommunicationIdentity` since I was intending this to be under the same namespace when in an SDK, but I can rename it to `CommuncationTurn`

Reply:
Operation IDs follow the `NOUN_VERB` pattern.  Using C# as the example, the noun is used to create a class and interface name, and the verb creates a method on said class/interface.  The namespace is specified elsewhere, usually in the readme.<language>.md file.

Sounds like perhaps this should be `CommunicationTurn_IssueCredentials`?

Reply:
So the NOUN is the class name? I'm a bit confused on that, since the existing class `CommunicationIdentityClient` has a method `IssueToken`, which is defined in the follow swagger
```
"/identities/{id}/token": {
      "post": {
        "tags": [
          "Token"
        ],
        "summary": "Generate a new token for an identity.",
        "operationId": "CommunicationIdentity_IssueToken",
        "consumes": [
          "application/json"
        ],
```

Shouldn't the NOUN be "CommunicationIdentityClient" then?

Also the intention for this `IssueTurnCredentials` method is to be on the `CommunicationIdentityClient` class. This API is defined outside the existing `CommunicationIdentity.json` swagger since it is an experimental API to be released in a separate PrivatePreview release of the ACS SDK, but without any changes to the existing Identity APIs. This swagger would then be combined with the `CommunicationIdentity.json` swagger  during SDK generation under the `Azure.Communication.Identity` namespace.

Reply:
Ah ok that makes sense for the naming then, thanks for the explanation.